registrations/signals.py

```diff
@@ -7,7 +7,9 @@
 
 
 def _signup_or_group_post_delete(instance: Union[SignUp, SignUpGroup]) -> None:
-    instance.send_notification(SignUpNotificationType.CANCELLATION)
+    contact_person = getattr(instance, "contact_person", None)
+    if contact_person:
+        contact_person.send_notification(SignUpNotificationType.CANCELLATION)
```

Comment:
Consider moving this as the last item in post_save? If the code below unexpectedly fails and the whole view transaction is rolled back,  the message would have already been sent.

Reply:
Moved this to the end of `post_delete`